packages/chevrotain/src/parse/parser/traits/recognizer_api.ts

```diff
@@ -59,7 +59,7 @@ export class RecognizerApi {
     this: MixedInParser,
     idx: number,
     actionORMethodDef: GrammarAction<OUT> | DSLMethodOpts<OUT>
-  ): OUT {
+  ): OUT | undefined {
```

Comment:
note to self: 
- how to align the public apis to this?
- do they need to be aligned?
- can we fix the strict null checks in a different way?